engine/evd.ml

```diff
@@ -1783,3 +1783,23 @@ let evars_of_filtered_evar_info (type a) evd (evi : a evar_info) =
        | Evar_empty -> Evar.Set.empty
        | Evar_defined b -> evars_of_term evd b)
        (evars_of_named_context evd (evar_filtered_context evi)))
+
+let drop_new_defined ~original sigma =
```

Comment:
In a typical non-evar-using abstract call, this should be basically a noop
We might be able to be even faster by assuming that new evars have higher index than the max bindings in the original map but probably not worth relying on such implicit invariants